### ISSUE TITLE
Some `alpha_complex_3d` tests fail on the s390x architecture

### ISSUE THREAD

**Issue by gspr** [GUDHI/gudhi-devel#120]
As part of upgrading the GUDHI Debian package to 3.0.0, I have discovered the following test failures on the s390x architecture (and only there):

```
19/179 Test  #19: Alpha_complex_3d_test_unit ..................................................................***Failed    0.09 sec
Fast alpha complex 3d
Exact alpha complex 3d

 20/179 Test  #20: Weighted_alpha_complex_3d_test_unit .........................................................***Failed    0.08 sec
Check exception throw in debug mode
Check exception throw in debug mode
Check exception throw in debug mode
Weighted alpha complex 3d from points and weights
Weighted alpha complex 3d from points and weights
Weighted alpha complex 3d from points and weights

 22/179 Test  #22: Weighted_periodic_alpha_complex_3d_test_unit ................................................***Failed    9.21 sec
Weighted periodic alpha complex 3d exception throw
Cuboid is not iso exception
0 <= point.weight() < 1/64 * domain_size * domain_size exception
wp_points and p_weights size exception
Weighted periodic alpha complex 3d exception throw
Cuboid is not iso exception
0 <= point.weight() < 1/64 * domain_size * domain_size exception
wp_points and p_weights size exception
Weighted periodic alpha complex 3d exception throw
Cuboid is not iso exception
0 <= point.weight() < 1/64 * domain_size * domain_size exception
wp_points and p_weights size exception
Fast weighted periodic alpha complex 3d
```

As far as I can tell, these tests were not present in 2.3.0. All 176 other tests pass.

I do not have direct access to s390x hardware, and I assume that that architecture is of as little importance to you as it is to me, but I thought I would let you know in case the failure is symptomatic of something else. (For the Debian package I will simply skip the tests on this architecture and file a self-bug for documentation purposes).

**Comment by mglisse** [comment]
Thanks. Without access to s390x hardware (or simulator?) to reproduce it, it will be hard to debug, and the issue is as likely to come from CGAL, so we will probably leave it alone for now, but it is still useful to know.

**Comment by gspr** [comment]
Marc Glisse <notifications@github.com> writes:

> Thanks. Without access to s390x hardware (or simulator?) to reproduce
> it, it will be hard to debug, and the issue is as likely to come from
> CGAL, so we will probably leave it alone for now, but it is still
> useful to know.

Of course, same on my side. I just thought I'd report it for
completeness' sake. For the Debian package I will just skip the tests
and label the functionality as broken on this architecture. In the
unlikely case that there are any users on s390x that get back to me, I
will report to you guys.

**Comment by mglisse** [comment]
I ran these tests (possibly with slightly different compiler options) using a cross-compiler and qemu, and they passed. The message pasted in the report gives a vague idea of where it stopped in the test, but not why (did it crash?).

**Comment by gspr** [comment]
Marc Glisse <notifications@github.com> writes:

> I ran these tests (possibly with slightly different compiler options)
> using a cross-compiler and qemu, and they passed. The message pasted
> in the report gives a vague idea of where it stopped in the test, but
> not why (did it crash?).

I haven't look at this since originally filing this issue during the
GUDHI 3.0.0 days – I'm guessing there aren't any actual GUDHI users on
s390x out there, so I haven't looked more into it.

After your comment, I retried the build-time tests on real s390x
hardware (a Debian porterbox), and everything is fine now. I'm guessing
it was really a CGAL bug that got fixed. The hordes of GUDHI users on
s390x are safe – we can close this issue ;-)
